File: python-pytest/abcmouse/factories/user_factory.py
from faker import Faker

from abcmouse.models.user import User


class UserFactory:
    def __init__(self):
        self.fake = Faker()
        pass

    @staticmethod
    def create_user(name=None):
        fake = Faker()
        if name:
            # If a name is provided, use it to create the email
            email = f"{name.replace(' ', '.').lower()}@{fake.domain_name().lower}"
        else:
            # Generate a random name and email using Faker
            name = fake.name()
            email = fake.email()

        # Create a User instance with the data
        return User(name, email)

File: CSharpNunitSelenium/CSharpNunitSelenium/ABCMouse/Factories/UserFactory.cs
using Bogus;
using CSharpNunitSelenium.ABCMouse.Models;

namespace CSharpNunitSelenium.ABCMouse.Factories;

public class UserFactory
{
    private readonly Faker<User> _faker;

    public UserFactory()
    {
        _faker = new Faker<User>()
            .RuleFor(u => u.Email, (f, u) => f.Internet.Email(u.Name));
    }

    public User Create(string? name = null)
    {
        if (string.IsNullOrEmpty(name))
        {
            return _faker.Generate();
        }
        else
        {
            return new User
            {
                Name = name,
                Email = _faker.Generate().Email
            };
        }
    }
}